bindings/java/src/main/native/endpoint.cc

```diff
@@ -129,7 +129,7 @@ Java_org_openucx_jucx_ucp_UcpEndpoint_closeNonBlockingNative(JNIEnv *env, jclass
     param.cb.send       = jucx_request_callback;
 
     ucs_status_ptr_t status = ucp_ep_close_nbx((ucp_ep_h)ep_ptr, &param);
-    process_request(env, jucx_request, status);
+    process_request(env, reinterpret_cast<jobject>(param.user_data), status);
```

Comment:
maybe pass `const ucp_request_param_t *param` to `process_request()`?
to make code more readable and ensure such leak would not be added any more

Reply:
Sounds good to me. I wasn't sure what is the best way to pass the reference to `process_request()`, but thinking of it again, avoiding the typecast seems better than my current solution.